src/components/ImageView/Overlay/OverlayComponent.tsx

```diff
@@ -169,7 +168,7 @@ export class OverlayComponent extends React.Component<OverlayComponentProps> {
         const distanceMeasuringColor = frame.distanceMeasuring?.color;
         const distanceMeasuringFontSize = frame.distanceMeasuring?.fontSize;
         const distanceMeasuringLineWidth = frame.distanceMeasuring?.lineWidth;
-        const title = this.props.overlaySettings.title.customText ? frame.titleCustomText : frame.filename;
+        const title = this.props.overlaySettings.title.customText ? this.props.image?.store?.titleCustomText : this.props.image?.store?.filename;
```

Comment:
Same question as above.

Reply:
We need to use `ColorBlendingStore.filename` instead of the base frame's filename.